kate/grid/src/dims.rs

```diff
@@ -0,0 +1,64 @@
+/// The dimensions of a grid
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Dimensions {
+	width: usize,
```

Comment:
These are specifically for talking about the size of in-memory arrays. You're better off using `usize` than using u32 and casting all over the place.

These types all need to get pulled together anyways, since there's about 4 different types for this.